platform/http_request.cpp

```diff
@@ -121,6 +121,7 @@ class MemoryHttpRequest : public HttpRequest, public IHttpThreadCallback
                     CallbackT onFinish, CallbackT onProgress)
     : HttpRequest(onFinish, onProgress), m_writer(m_downloadedData)
   {
+    m_progress = make_pair(0, 0);
```

Comment:
HttpRequest - низкоуровневый код, который работает так, как и задумано.
Проблема, которую пытается раскопать Илья, во взаимодействии стораджа с UI. Сторадж, который умеет работать с группами файлов, хранит свою копию прогреса, пытаясь ее как-то синхронизировать с низкоуровневым кодом, который работает с одним файлом. И делает это не совсем правильно. @bykoianko обещал посмотреть, проблема с высокой вероятностью где-то в storage.cpp.

> On Apr 7, 2016, at 15:54, Илья Гречухин <notifications@github.com> wrote:
> 
> In platform/http_request.cpp <https://github.com/mapsme/omim/pull/2756#discussion_r58865516>:
> 
> > @@ -121,6 +121,7 @@ class MemoryHttpRequest : public HttpRequest, public IHttpThreadCallback
> >                      CallbackT onFinish, CallbackT onProgress)
> >      : HttpRequest(onFinish, onProgress), m_writer(m_downloadedData)
> >    {
> > +    m_progress = make_pair(0, 0);
> Юра, в конструкторе HttpRequest прогресс делается не нулевым:
> m_progress(make_pair(0, -1))
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2756/files/193bf11a2e6f8a3ead0d909e7f91c3f351ef2e0a#r58865516>